src/launchpad/models/common.py

```diff
@@ -44,65 +51,79 @@ class SymbolInfo(BaseModel):
-        default_factory=dict, description="Files grouped by type/extension"
-    )
-    duplicate_files: List[DuplicateFileGroup] = Field(default_factory=list, description="Groups of duplicate files")
-    largest_files: List[FileInfo] = Field(default_factory=list, description="Largest files in the bundle")
+    files: List[FileInfo] = Field(..., description="List of all files in the bundle")
+
+    @property
+    def total_size(self) -> int:
+        """Total bundle size in bytes."""
+        return sum(f.size for f in self.files)
 
     @property
-    def total_duplicate_savings(self) -> int:
-        """Total potential savings from removing duplicates."""
-        return sum(group.potential_savings for group in self.duplicate_files)
+    def file_count(self) -> int:
+        """Total number of files."""
+        return len(self.files)
 
     @property
     def file_type_sizes(self) -> Dict[str, int]:
         """Total size by file type."""
-        return {file_type: sum(file.size for file in files) for file_type, files in self.files_by_type.items()}
+        return {file.file_type: file.size for file in self.files}
 
 
-class BaseAppInfo(BaseModel):
-    """Base app information that applies across platforms."""
+class FileInfo(BaseModel):
+    """Information about a file in the app bundle."""
 
     model_config = ConfigDict(frozen=True)
 
-    name: str = Field(..., description="App display name")
-    version: str = Field(..., description="App version")
-    build: str = Field(..., description="Build number")
+    path: str = Field(..., description="Relative path in the bundle")
+    size: int = Field(..., ge=0, description="File size in bytes")
+    file_type: str = Field(..., description="File type/extension")
+    hash_md5: str = Field(..., description="MD5 hash of file contents")
+    treemap_type: str = Field(..., description="Type for treemap visualization")
 
 
-class BaseBinaryAnalysis(BaseModel):
-    """Base binary analysis that applies across platforms."""
+class DuplicateFileGroup(BaseModel):
+    """Group of duplicate files."""
 
     model_config = ConfigDict(frozen=True)
 
-    executable_size: int = Field(..., ge=0, description="Main executable size in bytes")
-    architectures: List[str] = Field(..., description="CPU architectures")
-    linked_libraries: List[str] = Field(default_factory=list, description="Linked dynamic libraries")
-    sections: Dict[str, int] = Field(default_factory=dict, description="Binary sections and their sizes")
+    files: List[FileInfo] = Field(..., description="Files in the group")
+    potential_savings: int = Field(..., ge=0, description="Potential size savings in bytes")
 
+    @property
+    def duplicate_count(self) -> int:
+        """Number of duplicate files (excluding the original)."""
+        return len(self.files) - 1
 
-class BaseAnalysisResults(BaseModel):
-    """Base analysis results structure."""
+
+class BaseInsightResult(BaseModel):
```

Comment:
Might be good to move these into a `/models/insights/common.py` folder as I imagine there's going to be a lot of insights and they might bloat this file. Don't feel too strongly here but I envision this file getting pretty large if we dump all insights here (and the platform-specific results files too).